imports/api/shareddocs/methods.js

```diff
@@ -49,4 +49,7 @@ export const remove = new ValidatedMethod({
     Shareddocs.remove(_id);
   },
 });
-  
\ No newline at end of file
+
+export function cleanCanceledVoteAttachments() {
+  Shareddocs.remove({ $where: 'this.topicId === this.userId' });
```

Comment:
Azt irja a mongodb manual, hogy a $where egy nagyon rugalmas operacio, mert barmilyen javascriptet le lehet futtatni vele a kollekcio mindene elemere, viszont ez nagyon lassu is, mert vegig kell menni minden elemen.
A $expr sokkal gyorsabb, mert az a monngodb-n belul ki tudja optimalizalni, nem kell javascriptet futasson minden elemre egyesevel. Ezert inkabb azt javasolja ilyen esetekre.
Atirod inkabb arra?

Reply:
Először én is $expr-et használtam, de hibára futott. Mintha a meteoros mongo nem értette volna a parancsot, ezért váltottam $where-re. A pontos hibaüzenetre már nem emlékszem, de ha gondolod reprodukálom.

Reply:
Akkor erre hibat dobott?
Shareddocs.remove({ $expr: { $eq: ['$topicId', '$userId'] } });

Reply:
Igen: MongoError: unknown top level operator: $expr

Reply:
Aha. Ugy tunik a meteor a mongo 3.4-es verziojat hasznalja. Az $expr pedig a 3.6-ban lett elerheto. Ugyhogy addig nem tudjuk hasznalni, amig a meteor fel nem frissiti a mongo-jat, amit hasznal.

Reply:
Ha úgy van ahogy hagytam, akkor az ubuntu mongo-ját használja, nem a minu-mongo-t. Lehet a legújabbra frissíteni.